src/shogun/mathematics/linalg/LinalgBackendBase.h

```diff
@@ -28,69 +27,63 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  *
- * Authors: 2016 Pan Deng, Soumyajit De, Viktor Gal
+ * Authors: 2016 Pan Deng, Soumyajit De, Heiko Strathmann, Viktor Gal
  */
 
 #include <shogun/lib/config.h>
-#include <shogun/io/SGIO.h>
+#include <shogun/lib/common.h>
 #include <shogun/lib/SGVector.h>
-#include <shogun/mathematics/linalg/BaseVector.h>
+#include <shogun/io/SGIO.h>
+#include <shogun/mathematics/linalg/GPUMemoryBase.h>
 #include <memory>
 
-#ifndef GPU_Vector_H__
-#define GPU_Vector_H__
-
-#ifdef HAVE_CXX11
+#ifndef Linalg_Backend_Base_H__
+#define Linalg_Backend_Base_H__
 
 namespace shogun
 {
 
-/** GPU vector wrapper structure */
-template <class T>
-struct GPUVector : public BaseVector<T>
+/** Base class for generic linalg methods */
+class LinalgBackendBase
```

Comment:
Actually, for this to work with plugins, we need to be inheriting from CSGObject.
But for now, just leave it like this (you can remove the get_name)
@vigsterkr I guess we can easily change that later? Or even @sanuj can make this the first plugin